### ISSUE TITLE
CA1017: MarkAssembliesWithComVisible doesn't generate any diagnostic output

### ISSUE THREAD

**Issue by BlackStar42X** [dotnet/roslyn-analyzers#7007]
### Analyzer

**Diagnostic ID**: [CA1017](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1017): `Mark assemblies with ComVisibleAttribute`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 7.0.402](https://dotnet.microsoft.com/download/dotnet/7.0)

### Describe the bug

No diagnostic output is generated when the rule is enabled and violated.

### Steps To Reproduce

1. Create a new project in VS. In my case I picked VS 2022 and a class library.
1. When asked, pick .NET 6.0 as the target framework.
1. Add a `.globalconfig` file in project folder and add this line:
```
dotnet_diagnostic.CA1017.severity = warning
```
4. Run code analysis.

### Expected behavior

A diagnostic output about the missing ComVisibleAttribute.

### Actual behavior

No output.

**Comment by mavasani** [comment]
Couple of questions:
- Did you add `is_global = true` at top of your .globalconfig file?
- Are you able to see the diagnostic if you explicitly run a `Build` or `Rebuild` command? This will help us narrow down if the issue is with `Run code analysis` command or the core analyzer engine running the analyzers/reporting diagnostics.

Thanks

**Comment by BlackStar42X** [comment]
* I didn't include `is_global = true` at the top of the file. It was my understanding that this line in `.globalconfig` is for clarity only, so I didn't anticipate it to affect a diagnostic rule.
* The diagnostic also doesn't show up when I explicitly run `Build` or `Rebuild`.

**Comment by sharwell** [comment]
@mavasani does the following code consider only types in the global namespace?

https://github.com/dotnet/roslyn-analyzers/blob/5dc99c6249a4cf845fbdcc367641c1b32347b85a/src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/MarkAssembliesWithComVisible.cs#L92-L96

The tests don't include any examples with types inside a namespace.
